### PR TITLE
Add commands to build and push django container to AWS ECR

### PR DESCRIPTION
**Pull request by RishabhJain2018**
@deshraj Please review the PR.

### CODE REVIEW

File: requirements/dev.txt [hunk 1]
```diff
@@ -9,6 +9,7 @@ django-extensions==1.7.8
 django-silk==1.0.0
 django-spaghetti-and-meatballs==0.2.2
 flake8==3.0.4
+google-compute-engine==2.8.3
```
Comment (deshraj) [inline requirements/dev.txt]: Why do we need this?

File: settings/test.py [hunk 2]
```diff
@@ -22,4 +22,5 @@
     }
 }
 
+DEV = False
```
Comment (deshraj) [inline settings/test.py]: We don't need a new variable `DEV`. 

`settings.DEBUG = True` means dev environment only.

File: requirements/dev.txt (hunk 1 above)
Comment (RishabhJain2018) [reply requirements/dev.txt]: Earlier it threw that this module isn't found. I am going to remove this line only.

File: settings/test.py (hunk 2 above)
Comment (RishabhJain2018) [reply settings/test.py]: Please check the condition  https://github.com/Cloud-CV/EvalAI/pull/1792/files#diff-aa6d983093bc40e4ce5d8205f511db8dR78

Comment (deshraj) [reply settings/test.py]: This variable will cause confusion for no reason. We already have the `TEST` variable which you can make use of in the code snippet.

File: evalai/urls.py [hunk 3]
```diff
@@ -75,7 +75,7 @@
                ]
 
 # DJANGO-SPAGHETTI-AND-MEATBALLS URLs available during development only.
-if settings.DEBUG:
+if settings.DEBUG or not settings.TEST:
```
Comment (deshraj) [inline evalai/urls.py]: I am not sure if we need an additional check here.

File: scripts/deploy/production.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
+BRANCH="production"
```
Comment (deshraj) [inline scripts/deploy/production.sh]: We don't need this script now, right?

File: .travis.yml [hunk 5]
```diff
@@ -39,7 +45,9 @@ script:
 
 after_success:
   - coveralls --rcfile=.coveragerc
-
+  - pip install awscli
+  - bash scripts/deploy/staging.sh
+  - bash scripts/deploy/production.sh
```
Comment (deshraj) [inline .travis.yml]: We dont need the two scripts now, right?

File: docker-compose-staging.yml [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+  worker:
+    image: ${AWS_ACCOUNT_ID}.dkr.ecr.us-east-1.amazonaws.com/evalai-staging-worker:${TRAVIS_COMMIT}
+    build:
+      context: ./
+      dockerfile: docker/prod/worker/Dockerfile
+    depends_on:
+      - django
+    environment:
+      # TODO: Sort these entries
```
Comment (deshraj) [inline docker-compose-staging.yml]: Please remove this.

File: docker-compose-staging.yml [hunk 7]
```diff
@@ -0,0 +1,76 @@
+version: "3"
+services:
+
+  django:
+    image: ${AWS_ACCOUNT_ID}.dkr.ecr.us-east-1.amazonaws.com/evalai-staging-backend:${TRAVIS_COMMIT}
+    container_name: django
+    environment:
+      # TODO: Sort these entries
```
Comment (deshraj) [inline docker-compose-staging.yml]: Please remove this line.

File: scripts/deploy/deploy.sh [hunk 8]
```diff
@@ -0,0 +1,35 @@
+if [ "$TRAVIS_PULL_REQUEST" != "false" -o "$TRAVIS_BRANCH" != "staging" -o "$TRAVIS_BRANCH" != "production" ]; then
+    echo "Skipping deploy to staging or production server; The request or commit is not on staging or production branch"
+    exit 0
+fi
+
+if [ "$TRAVIS_PULL_REQUEST" != "false" -a "TRAVIS_BRANCH" != "staging"]; then
```
Comment (deshraj) [inline scripts/deploy/deploy.sh]: Please modify line 6-30 and put them in a function(). The logic could be something like this

```
if branch == staging:
    deploy to evalai-staging-* ECR repos with staging credentials
else if branch == production:
   deploy to evalai-* ECR repos with prod credentials
```

**Pull request merged**